test/testdata/goldens/directory_option_test/commands taking a --directory~-C parameter work.txt

```diff
@@ -17,7 +17,7 @@ Changed 1 dependency in myapp!
 -------------------------------- END OF OUTPUT ---------------------------------
 
 ## Section 2
-$ pub -C myapp/example get --directory=myapp bar
+$ pub -C 'myapp/example' get --directory=myapp bar
```

Comment:
Should we do a tweak to escaping mechanism, so that we don't need to escape when there is a `/` or `.` in the string... those are generally safe in bash, right? (and quite common in our tests)